h2o-core/src/main/java/water/FailedNodeWatchdogExtension.java

```diff
@@ -169,28 +172,42 @@ public void run() {
      * Helper MR task used to detect clientDisconnectedConsensus on the timeout we last heard from the watchdog client
      */
     private static class WatchdogClientDisconnectedTask extends MRTask<WatchdogClientDisconnectedTask> {
-        private boolean clientDisconnectedConsensus = false;
-        private H2ONode clientNode;
-        private long watchdogClientRetryTimeout;
+        private final H2ONode clientNode;
+        private final long watchdogClientRetryTimeout;
+
+        // OUT
+        private boolean[] clientConnected;
+
         WatchdogClientDisconnectedTask(H2ONode clientNode, long  watchdogClientRetryTimeout) {
             this.clientNode = clientNode;
             this.watchdogClientRetryTimeout = watchdogClientRetryTimeout;
         }
 
         @Override
         public void reduce(WatchdogClientDisconnectedTask mrt) {
-            this.clientDisconnectedConsensus = this.clientDisconnectedConsensus && mrt.clientDisconnectedConsensus;
+            for (int i = 0; i < clientConnected.length; i++) {
+                clientConnected[i] &= mrt.clientConnected[i];
+            }
         }
 
         @Override
         protected void setupLocal() {
+            clientConnected = new boolean[H2O.getCloudSize()];
```

Comment:
Just a question: Is the number of nodes final at this point?

Reply:
Yes, it should be. The cloud will be locked. The reason why it is final is that we actually do see a client. Because the client is run only after the cloud is fully formed we can relly on the fact the list of nodes won't change.

Reply:
Ok, perfect. Thanks for the explanation!